testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,5 +180,53 @@ public void testLateralViewJsonTupleAs()
                 "SELECT * FROM test_json_tuple_view",
                 queryAssert -> queryAssert.containsOnly(row(3, "Mateusz", "Gajewski", "true", "1000", null, null)));
     }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionViews()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_view");
+        onHive().executeQuery("DROP VIEW IF EXISTS union_distinct_view");
+
+        Map<String, List<QueryAssert.Row>> expected = new HashMap<>();
+
+        // UNION ALL
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION ALL\n"
+                + "SELECT n_regionkey FROM nation\n");
```

Comment:
let's have a source with some duplicates, to verify all duplicates are retained (no DISTINCT over any of the sources individually)